.github/workflows/build-test.yml

```diff
@@ -0,0 +1,42 @@
+name: FPP CI
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+permissions:
+  contents: read
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: Set up JDK 11
+      uses: actions/setup-java@v3
+      with:
+        java-version: '11'
+        distribution: 'temurin'
+        cache: 'sbt'
+    - name: Install Scala
+      run: |
+        curl -fL https://github.com/coursier/launchers/raw/master/cs-x86_64-pc-linux.gz | gzip -d > cs
+        chmod +x cs
+        echo Y | ./cs setup
+    - name: Install fpp tools
+      run: compiler/install
+    - name: Run sbt tests
+      run: sbt test
```

Comment:
Should we run `cd compiler` before `sbt test`? It looks like the run is warning that it can't find `project/sbt-properties`. If we run `sbt test` in `compiler`, that warning should disappear.

Reply:
Ok, I made that change.